lightclient/src/rpc_to_internal.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2019-2024 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use anyhow::Context;
+use base64::{Engine as _, engine::general_purpose};
+use filecoin_f3_certs as certs;
+use filecoin_f3_gpbft::chain;
+use filecoin_f3_gpbft::{BitField, Cid, ECChain, PubKey, SupplementalData};
+use filecoin_f3_rpc::{self as rpc};
+use keccak_hash::H256;
+
+/// Converts RPC power entry to the internal format
+pub fn convert_power_entry(val: rpc::PowerEntry) -> anyhow::Result<filecoin_f3_gpbft::PowerEntry> {
+    Ok(filecoin_f3_gpbft::PowerEntry {
+        id: val.id,
+        power: val.power,
+        pub_key: PubKey(general_purpose::STANDARD.decode(&val.pub_key)?), // base64 -> bytes -> PubKey
+    })
+}
+
+/// Converts RPC finality certificate to the internal format
+pub fn convert_certificate(
+    val: rpc::FinalityCertificate,
+) -> anyhow::Result<certs::FinalityCertificate> {
+    Ok(certs::FinalityCertificate {
+        gpbft_instance: val.instance,
+        ec_chain: convert_chain(val.ec_chain)?,
+        supplemental_data: SupplementalData {
+            commitments: Default::default(),
+            power_table: Cid::try_from(val.supplemental_data.power_table.cid.as_str())?,
+        },
+        power_table_delta: val
+            .power_table_delta
+            .into_iter()
+            .map(convert_power_table_delta)
+            .collect::<anyhow::Result<Vec<_>, _>>()?,
+        signers: convert_bitfield(val.signers)?,
+        signature: general_purpose::STANDARD.decode(&val.signature)?, // base64 -> bytes
+    })
+}
+
+fn convert_chain(val: Vec<rpc::ECTipSet>) -> anyhow::Result<ECChain> {
+    if val.is_empty() {
+        return Err(anyhow::anyhow!("empty ec_chain"));
+    }
```

Comment:
```suggestion
    ensure!(!val_is_empty(), "empty ec_chain");
```
nit: use https://docs.rs/anyhow/latest/anyhow/macro.ensure.html for  such constructs.

Reply:
Fixed here: https://github.com/ChainSafe/rust-f3/pull/27/commits/95db86a2e4fad4b5a8246a53980cd28fd254eeb5